scapy/utils.py

```diff
@@ -1905,6 +1905,250 @@ def write_packet(self,
         )
 
 
+@conf.commands.register
+def rderf(filename, count=-1):
+    # type: (Union[IO[bytes], str], int) -> PacketList
+    """Read a ERF file and return a packet list
+
+    :param count: read only <count> packets
+    """
+    with ERFEthernetReader(filename) as fdesc:
+        return fdesc.read_all(count=count)
+
+
+class ERFEthernetReader(_SuperSocket):
```

Comment:
Why are you inheriting from `_SuperSocket` instead of `RawPcapReader` ? It seems possible to resuse some code.